### DISCUSSION TITLE
[Help] Is it possible to evaluate attributes of an object lazy at runtime via a callback e.g.

### DISCUSSION THREAD

**Discussion by sassman** [mitsuhiko/minijinja#214] [Q&A] [answered]
in the [docs](https://docs.rs/minijinja/latest/minijinja/syntax/index.html#other-operators) there is:

> . / []: Get an attribute of an object.

I wonder if there is a way to somehow lazy (e.g. similar to `add_function`) evaluate attributes of an object.
So that I could write something like below, to refer to the equivalent of `std::env::var("USER")`:

```
{{ env.USER }}
```

So the main thing I'd like to achieve is not to have to pre-allocate an object `env` with all the attributes that are the actual process environment variables. I know that registering a `HashMap` would actually work.

**Comment by sassman** [comment, marked as answer]
I just came across this: https://github.com/mitsuhiko/minijinja/issues/160#issuecomment-1331299303 
It might be possible to solve the use case by a custom `StructObject` impl where `get_field` could take care of it, right?

**Reply by mitsuhiko** [reply]
Correcty, you can implement your own `StructObject`. You find an example here: https://github.com/mitsuhiko/minijinja/blob/main/examples/load-lazy/src/main.rs

**Reply by sassman** [reply]
Thanks, works like charm.
